### PR TITLE
Reorganize project to be a more idiomatic use of workspace, proto, and docker

### LINKED ISSUE #10: move fluent bit to namespace
- currently in default namespace

### LINKED ISSUE #17: Idiomatic project structure, particularly for docker and protos
Each separate crate in the workspace has its own copy of the proto file, as if we were building clients for someone else's services and to support docker build from the directory. We need to come up with an approach that makes more sense, probably move the docker files to the top of the workspace and have each crate service depend on the generated grpc lib, building from the proto file.

### PR DESCRIPTION
**Pull request by DanMass**
# Type of PR

- [x] Documentation changes
- [x] Code changes
- [ ] Test changes
- [x] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [x] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR
The primary purpose of this PR is to refactor toward a more idiomatic combination of rust apps in a workspace, docker, and grpc. There is also some name cleanup and convenience help for the linkerd image pulling issues in the cluster.

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [ ] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes #17
- Closes #10

### CODE REVIEW

File: deploy/fluentbit/stdout-config.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ data:
     [FILTER]
         Name                kubernetes
         Match               kube.*
-        Kube_URL            https://kubernetes.default.svc:443
+        Kube_URL            https://kubernetes.fluentbit.svc:443
```
Comment (bartr) [inline deploy/fluentbit/stdout-config.yaml:42]: I'm not sure this is right as you want the k8s api service

Comment (DanMass) [reply deploy/fluentbit/stdout-config.yaml:42]: Good point. I did a replace and didn't see that this one wasn't like the others.

File: deploy/fluentbit/fluentbit-pod.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 kind: Pod
 metadata:
   name: fluentb
-  namespace: default
+  namespace: fluentbit
```
Comment (bartr) [inline deploy/fluentbit/fluentbit-pod.yaml:5]: should we use the logging namespace?

**Review by bartr — APPROVED** [review]
approved with comments

File: deploy/fluentbit/fluentbit-pod.yaml (hunk 2 above)
Comment (DanMass) [reply deploy/fluentbit/fluentbit-pod.yaml:5]: Good question. I was going with the patten of other tools in there now, which is to name for the project, not the function.

Comment (DanMass) [reply deploy/fluentbit/fluentbit-pod.yaml:5]: What do you think?

Comment (bartr) [reply deploy/fluentbit/fluentbit-pod.yaml:5]: I've seen both. We use monitoring for prometheus and grafana so logging feels better to me. But not by much. I'm OK with either.

Comment (DanMass) [reply deploy/fluentbit/fluentbit-pod.yaml:5]: Then I'll merge this and we can think about it. Given that linkerd comes with its prometheus and grafana set ups, it won't be that odd in this container.

Comment (DanMass) [reply deploy/fluentbit/fluentbit-pod.yaml:5]: Thank you!

**Pull request merged**